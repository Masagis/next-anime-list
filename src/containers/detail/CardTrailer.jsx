import React from 'react'

function CardTrailer(props) {
  const { data } = props
  return (
    <div>
      {data?.trailer?.embed_url === null ? (
        <div className="text-center" />
      ) : (
        <div className="video-responsive mb-3">
          <iframe
            width="100%"
            height="480"
            src={`${data?.trailer?.embed_url}`}
            frameBorder="0"
            allow="clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="Embedded youtube"
          />
        </div>
      )}
    </div>
  )
}

export default CardTrailer
