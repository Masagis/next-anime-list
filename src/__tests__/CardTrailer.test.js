/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import CardTrailer from '../containers/detail/CardTrailer'

describe('Component CardTrailer', () => {
  test('should be render the component card recomendation anime by id', () => {
    const data = {
      trailer: {
        youtube_id: 'l_98K4_6UQ0',
        url: 'https://www.youtube.com/watch?v=l_98K4_6UQ0',
        embed_url:
          'https://www.youtube.com/embed/l_98K4_6UQ0?enablejsapi=1&wmode=opaque&autoplay=1',
        images: {
          image_url: 'https://img.youtube.com/vi/l_98K4_6UQ0/default.jpg',
          small_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/sddefault.jpg',
          medium_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/mqdefault.jpg',
          large_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/hqdefault.jpg',
          maximum_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/maxresdefault.jpg',
        },
      },
    }

    render(<CardTrailer data={data} />)
    expect(screen.getByTestId('card-trailer-anime')).toBeInTheDocument()
    expect(screen.getByTestId('video-trailer-anime')).toHaveAttribute(
      'src',
      'https://www.youtube.com/embed/l_98K4_6UQ0?enablejsapi=1&wmode=opaque&autoplay=1',
    )
  })

  test('should be not render the component card trailer anime by id', () => {
    const data = {
      trailer: {
        youtube_id: null,
        url: null,
        embed_url: null,
        images: {
          image_url: null,
          small_image_url: null,
          medium_image_url: null,
          large_image_url: null,
          maximum_image_url: null,
        },
      },
    }

    render(<CardTrailer data={data} />)
    expect(screen.queryByTestId('hidden-trailer-anime')).not.toBeNull()
    expect(screen.queryByTestId('unhidden-trailer-anime')).toBeNull()
  })

  test('snapshoot', () => {
    const data = {
      trailer: {
        youtube_id: 'l_98K4_6UQ0',
        url: 'https://www.youtube.com/watch?v=l_98K4_6UQ0',
        embed_url:
          'https://www.youtube.com/embed/l_98K4_6UQ0?enablejsapi=1&wmode=opaque&autoplay=1',
        images: {
          image_url: 'https://img.youtube.com/vi/l_98K4_6UQ0/default.jpg',
          small_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/sddefault.jpg',
          medium_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/mqdefault.jpg',
          large_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/hqdefault.jpg',
          maximum_image_url:
            'https://img.youtube.com/vi/l_98K4_6UQ0/maxresdefault.jpg',
        },
      },
    }

    const renderComponent = render(<CardTrailer data={data} />)
    expect(renderComponent).toMatchSnapshot()
  })
})
