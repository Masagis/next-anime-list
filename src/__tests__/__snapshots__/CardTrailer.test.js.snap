// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CardTrailer snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="card-trailer-anime"
      >
        <div
          class="video-responsive mb-3"
          data-testid="unhidden-trailer-anime"
        >
          <iframe
            allow="clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen=""
            data-testid="video-trailer-anime"
            frameborder="0"
            height="480"
            src="https://www.youtube.com/embed/l_98K4_6UQ0?enablejsapi=1&wmode=opaque&autoplay=1"
            title="Embedded youtube"
            width="100%"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="card-trailer-anime"
    >
      <div
        class="video-responsive mb-3"
        data-testid="unhidden-trailer-anime"
      >
        <iframe
          allow="clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowfullscreen=""
          data-testid="video-trailer-anime"
          frameborder="0"
          height="480"
          src="https://www.youtube.com/embed/l_98K4_6UQ0?enablejsapi=1&wmode=opaque&autoplay=1"
          title="Embedded youtube"
          width="100%"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
